source/draw/gpu/opengl/backend/gles3/Gles3Context.ooc

```diff
@@ -191,25 +85,13 @@ Gles3Context: class extends GLContext {
 	createIndexBufferObject: override func (vertices: FloatPoint3D[], textureCoordinates: FloatPoint2D[], indices: IntPoint3D[]) -> GLIndexBufferObject {
 		Gles3IndexBufferObject new(vertices, textureCoordinates, indices)
 	}
-	validate: static func (value, expectedValue: UInt, function: String) {
-		if (value != expectedValue)
-			Debug error(function << " failed! Expected status %u but got %u. eglError=%d" format(expectedValue, value, eglGetError()))
-	}
-	validate: static func ~expression (success: Bool, function: String) {
-		if (!success)
-			Debug error(function << " failed with eglError=%d" format(eglGetError()))
-	}
 	create: static func ~shared (display: Pointer, nativeBackend: Long, sharedContext: This = null) -> This {
 		version(debugGL) { Debug print("Creating OpenGL Context") }
-		result := This new()
-		result _generate(display, nativeBackend, sharedContext) ? result : null
+		result := This new(EglDisplayContext new(display, nativeBackend, sharedContext getDisplayContextSafely()))
 	}
 	create: static func ~pbufferShared (sharedContext: This = null) -> This {
 		version(debugGL) { Debug print("Creating OpenGL Context") }
-		result := This new()
-		result _generate(sharedContext) ? result : null
+		result := This new(EglDisplayContext new(sharedContext getDisplayContextSafely()))
```

Comment:
This looks cleaner but I do not know if this will work when GLES3 has to fall back to GLES2 since the constructor cannot have a boolean return value.